### ISSUE TITLE
Associate Customer with Cart doesn't work.

### ISSUE THREAD

**Issue by Deroswent** [medusajs/medusa#9629]
### Describe the bug

According to Create Cart in Storefront [documentation](https://docs.medusajs.com/v2/resources/storefront-development/cart/create), a cart can be associated with a user (last part of the documentation). But this doesn't work and conflicts with the API reference.   

If you try to pass both region_id and customer_id in the request body (as described in the last part of the documentation) - you will get the error “message”: “An unknown error occurred.”.       
Also, the reference API for [POST /store/carts ](https://docs.medusajs.com/v2/api/store#carts_postcarts) doesn't mention customer_id at all. So one of these articles is definitely incomplete or false. 

### System information

Medusa version (including plugins):  v2  RC-6   (2.0.0-rc-20241016093905) 
Node.js version: v20.9.0 (LTS)
Database: postgresql 16
Operating system: Windows 11
Browser (if relevant): no matter

### Steps to reproduce the behavior

1. Send POST /store/carts  with body, contains only region_id, like

```
 curl -X POST '{backend_url}/store/carts' \
-H 'Content-Type: application/json' \
-H 'x-publishable-api-key: <your_key>' \
--data-raw '{
  "region_id": "reg_01JABVFRWJK4H6YX9G9SF3Y235"
}'
```
it works, but the cart is not associated with a user

2. Now, send POST /store/carts  with body, contains region_id and customer_id as described in the documentation. Like:

```
 curl -X POST '{backend_url}/store/carts' \
-H 'Content-Type: application/json' \
-H 'x-publishable-api-key: <your_key>' \
--data-raw '{
  "region_id": "reg_01JABVFRWJK4H6YX9G9SF3Y235",
  "customer_id": "cus_01JABYG231BTGX7D2VDQMXRAPS"
}'
```

you'll receive error

```
{
    "code": "unknown_error",
    "type": "unknown_error",
    "message": "An unknown error occurred."
}
```

### Expected behavior

A shopping cart is expected to be created and associated with a customer

### Additional context

in the server console you can see a log detailing the server-side error

```
::1 - - [17/Oct/2024:03:09:01 +0000] "POST /store/carts HTTP/1.1" 500 86 "-" "PostmanRuntime/7.42.0"

error:   [
  {
    "code": "unrecognized_keys",
    "keys": [
      "customer_id"
    ],
    "path": [],
    "message": "Unrecognized key(s) in object: 'customer_id'"
  }
]
ZodError: [
  {
    "code": "unrecognized_keys",
    "keys": [
      "customer_id"
    ],
    "path": [],
    "message": "Unrecognized key(s) in object: 'customer_id'"
  }
]
    at Object.get error [as error] (C:\Users\deros\medusa\rc6\node_modules\@medusajs\medusa\node_modules\zod\lib\types.js:43:31)
    at ZodObject.parseAsync (C:\Users\deros\medusa\rc6\node_modules\@medusajs\medusa\node_modules\zod\lib\types.js:166:22)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async zodValidator (C:\Users\deros\medusa\rc6\node_modules\@medusajs\framework\src\zod\zod-helpers.ts:123:12)
    at async validateBody (C:\Users\deros\medusa\rc6\node_modules\@medusajs\framework\src\http\utils\validate-body.ts:30:27) {
  issues: [
    {
      code: 'unrecognized_keys',
      keys: [Array],
      path: [],
      message: "Unrecognized key(s) in object: 'customer_id'"
    }
  ],
  addIssue: [Function (anonymous)],
  addIssues: [Function (anonymous)],
  errors: [
    {
      code: 'unrecognized_keys',
      keys: [Array],
      path: [],
      message: "Unrecognized key(s) in object: 'customer_id'"
    }
  ]
}
```

cusomer_id is 100% correct, customer exist, registered, has account

![2024-10-17_06-49](https://github.com/user-attachments/assets/fc72632e-d2f9-4782-9899-925b3dcc6d82)

**Comment by VariableVic** [comment]
Hey @Deroswent, thanks for the report!

This mechanism has changed a bit, and that part of the docs is outdated. Carts are now automatically associated to the logged-in customer on creation. This is handled by the server - you no longer have to pass in `customer_id`.

On cart _update_, this mechanism isn't properly implemented yet, so you might experience some issues when updating a cart that was previously linked to the logged in customer. We're looking into this.

We'll update the docs to reflect the new situation.

If you have any questions or issues, feel free to re-open!

**Comment by bhuvan0616** [comment]
@VariableVic  
This issue still exists. 

- I have freshly installed medusa v2 with storefront.
- added item to cart.
- registered new user
- checkout from cart page
- Entering shipping address and click on continue to delivery.

Now new guest user with same email address  got created. and we can see that in admin dashboard.
![image](https://github.com/user-attachments/assets/47cd5327-0e77-4349-b971-5e3e0e76b678)

**Comment by Deroswent** [comment]
Confirmed, the problem exists. As @VariableVic said :

` Carts are now automatically associated to the logged-in customer on creation.`

But it doesn't seem to be working at all. I've tried passing both JWT and connect.sid (session cookie) when creating the cart, but it doesn't work and the cart doesn't bind to the user.

@VariableVic  could you please explain how to now bind a cart to a user?
What does “automatically associated” mean and how to do it using Store API?  According to the documentation, Store API [Create Cart](https://docs.medusajs.com/v2/api/store#carts_postcarts) still doesn't have any parameters to bind a user to the cart.

**Comment by bhuvan0616** [comment]
The PR "Add customer validation on cart update #9662" released on Medusa 2.0 Release Candidate 8. Then the PR was reverted on v2 release #9724. What the feature will be available @VariableVic?
